include/flecs/private/api_internals.h

```diff
@@ -333,6 +333,86 @@ FLECS_ALWAYS_INLINE ecs_table_t *flecs_table_traverse_add(
     ecs_id_t *id_ptr,
     ecs_table_diff_t *diff);
 
+#ifdef FLECS_SAFETY_LOCKS
+
+FLECS_API
+FLECS_ALWAYS_INLINE bool flecs_sparse_id_record_lock_read_begin(
```

Comment:
Since all these functions accept a component record, shouldn't this be in storage/component_index.h?